src/domain/usecases/ExportUsersUseCase.ts

```diff
@@ -0,0 +1,154 @@
+                );
+                const header = columns.map(this.getColumnNameFromProperty);
+                const table = [header, ...userRows];
+                return Papa.unparse(table);
+            }
+        }
+    }
+
+    private getColumnNameFromProperty(property: ColumnMappingKeys): string {
+        return columnNameFromPropertyMapping[property] || property;
+    }
+
+    private formatDate(stringDate?: Date | null): string | undefined {
+        return stringDate ? moment(stringDate).format("YYYY-MM-DD HH:mm:ss") : undefined;
+    }
+
+    private namesFromCollection(
+        collection: User[
+            | "userRoles"
+            | "userGroups"
+            | "organisationUnits"
+            | "dataViewOrganisationUnits"
+            | "searchOrganisationsUnits"],
+        field: OrgUnitKey | typeof defaultNameField,
+        toString: boolean
+    ): string | string[] {
+        // Check if field is in the collection, fallback on default name field
+        const nameField = _.some(collection, field) ? field : defaultNameField;
+        const namesArray = _(collection).map(nameField).value();
+
+        return toString ? namesArray.join(fieldSplitChar) : namesArray;
+    }
+
+    private getPlainUser(
+        user: User,
+        columns: ColumnMappingKeys[],
+        orgUnitsField: OrgUnitKey,
+        toString: boolean
+    ): Record<ColumnMappingKeys, typeof user[ColumnMappingKeys] | string[]> {
+        return _.pick(
+            {
+                ...user,
+                lastUpdated: this.formatDate(user.lastUpdated),
+                lastLogin: this.formatDate(user.lastLogin),
+                created: this.formatDate(user.created),
+                userRoles: this.namesFromCollection(user.userRoles, defaultNameField, toString),
+                userGroups: this.namesFromCollection(user.userGroups, defaultNameField, toString),
+                organisationUnits: this.namesFromCollection(user.organisationUnits, orgUnitsField, toString),
+                dataViewOrganisationUnits: this.namesFromCollection(
+                    user.dataViewOrganisationUnits,
+                    orgUnitsField,
+                    toString
+                ),
+                searchOrganisationsUnits: this.namesFromCollection(
+                    user.searchOrganisationsUnits,
+                    orgUnitsField,
+                    toString
+                ),
+                createdBy: user.createdBy?.username,
+                lastModifiedBy: user.lastModifiedBy?.username,
+            },
+            columns
+        );
+    }
+}
+
+export type ColumnMappingKeys = keyof typeof columnNameFromPropertyMapping;
+
+export type AllowedExportFormat = "json" | "csv";
+
+export type ExportUsersUseCaseOptions = {
+    name: string;
+    columns: ColumnMappingKeys[];
+    filterOptions?: ListOptions;
+    format: AllowedExportFormat;
+    orgUnitsField: OrgUnitKey;
+    isEmptyTemplate?: boolean;
```

Comment:
Let's try to avoid optional types. `isEmptyTemplate: boolean`.

Reply:
I initially tried to specified `isEmptyTemplate: boolean = false` but ESlint fixed it in `isEmptyTemplate = false`, like it was explicit. I've update it with default value inside the function instead in params